File: selectionsort/sort.js
function findSmallest(arr){
  // store the smallest element
  let smallest = arr[0];

  // store the index of the smallest el
  let smallest_index = 0;

  for(let i = 1; i < arr.length; i++){
    if(arr[i] < smallest){
      smallest = arr[i];
      smallest_index = i;
    }
  }

  return smallest_index;
}

// SELECTION SORT
function selectionSort(arr){
  let newArr = [];

  while(arr.length > 0){
    let smallest = findSmallest(arr);
    newArr.push(arr.splice(smallest, 1)[0]);
  }

  return newArr
}

// Measure execution time
console.time("SelectionSortTime"); // Start the timer
let test = selectionSort([5, 3, 6, 2, 10]);
console.timeEnd("SelectionSortTime"); // End the timer and log the time

console.log(test);
console.log(findSmallest(test));

File: selectionsort/sort.py
def findSmallest(arr):
  # stores the smallest value
  smallest = arr[0]

  # stores the index of the smallest value
  smallest_index = 0

  for i in range(1, len(arr)):
    if arr[i] < smallest:
      smallest = arr[i]
      smallest_index = i
  return smallest_index

#  SELECTION SORT 
def selectionSort(arr):
  newArr = []

  for i in range(len(arr)):
    # find the smallest element in the array and add it to the new array
    smallest = findSmallest(arr)
    newArr.append(arr.pop(smallest))
  return newArr


test = selectionSort([5, 3, 6, 2, 10])
print(test)
print(findSmallest(test))